services/auth/token/lambdas/generateToken.js

```diff
@@ -5,21 +5,39 @@ import * as response from '../../../../libs/response';
 import { signToken, verifyToken } from '../../../../libs/token';
 import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 import tokenValidationSchema from '../helpers/schema';
+import { logWarn } from '../../../../libs/logs';
 
 const CONFIG_AUTH_SECRETS = config.auth.secrets;
 const ACCESS_TOKEN_EXPIRES_IN_MINUTES = 20;
 const REFRESH_TOKEN_EXPIRES_IN_MINUTES = 30;
 
-export const main = async event => {
+export const main = async (event, context) => {
+  console.log(event);
   const [parseJsonError, parsedJson] = await to(parseJson(event.body));
+
   if (parseJsonError) {
+    logWarn(
+      'JSON Parse error',
+      context.awsRequestId,
+      'service-auth-token-generateToken-001',
+      parseJsonError
+    );
+
     return response.failure(parseJsonError);
   }
 
   const [validationError, validatedEventBody] = await to(
     validateEventBody(parsedJson, tokenValidationSchema)
   );
+
   if (validationError) {
+    logWarn(
+      'Validation error',
+      context.awsRequestId,
+      'service-auth-token-generateToken-002',
```

Comment:
Also curios on the thought behind the errorCode, what is the naming convention based upon and what's the purpose? easier to find/search for in the logs?

Reply:
Its generally used to distinguish error logs with the same error message to make it easy to find in code.
The convention here is a bit over the top with this template ${folder-names}-${filename}-${unique-number}

Reply:
That is correct.
I will look into automating it. Would make it easier to use, this would probably be a typical copy paste code snippet and it might be easy to forget to change or missunderstand its format/purpose.